### ISSUE TITLE
SourceMap's "sources" property is incorrect when using .tsx files

### ISSUE THREAD

**Issue by predeekc** [ivogabe/gulp-typescript#405]
I've been running into problems when using gulp-typescript followed by gulp-babel. If I logged the source maps after each step of the transform I noticed that the sources and file properties were the same when I used a .tsx file, but were correct (.ts and .js) when using .ts files.

This seems to get fixed up when using gulp-sourcemaps's .write method and isn't a problem when only gulp-typescript is used.  If I attach the gulp-babel plugin, it takes in the .js file and doesn't seem to match the chain of source maps properly and starts over.

It looks like the [line changing the extension from .ts to .js](https://github.com/ivogabe/gulp-typescript/blob/master/lib/output.ts#L112) might be the problem.  If I change the regex to /\.tsx?$/ it works fine as the .tsx extension is changed to .ts.

```
file.sourceMap.file = path.relative(file.sourceMap.sourceRoot, originalFileName).replace(/\.ts$/, '.' + jsExtension);
```

The scenario is have is compiling a src/root.tsx file to src/root.js. Here is the source map output before the change to the regex:
```
{
  "version":3,
  "sources":["src/root.js"],
  "names":[],
  "mappings":"...",
  "file":"src/root.js",
  "sourcesContent":["..."],
  "sourceRoot":"/src/"
}
```

Here is the source map output after the change to the regex:
```
{
  "version":3,
  "sources":["src/root.tsx"],
  "names":[],
  "mappings":"...",
  "file":"src/root.js",
  "sourcesContent":["..."],
  "sourceRoot":"/src/"
}
```

I've tried to reproduce the problem in the tests within gulp-typescript, but I've been unsuccessful. I have verified the change doesn't break any existing tests though.

If this is helpful and doesn't cause any other issues you know of, I can put together a pull request with this change.

**Comment by ivogabe** [comment]
Thanks for the debugging! I think that a more robust fix would be better. For instance, the source file could also be a `.jsx` file (with `allowJs: true`). `lib/utils.ts` has a function `splitExtension` that gives the part of the filename before the extension.

**Comment by ivogabe** [comment]
Can you create a pull request?

**Comment by predeekc** [comment]
Absolutely, I'll take a look at lib/utils.ts and put a PR together.

**Comment by predeekc** [comment]
Added PR #411 that uses utils.splitExtension instead of a regex to change the file extension

**Comment by ivogabe** [comment]
Fixed in #411.
